### PR TITLE
[macOS] Move scrollWheel event forwarding to a new class: NestableScrollView

### PR DESCRIPTION
**Pull request by alanjrogers**
This will let us re-use it elsewhere.

### CODE REVIEW

File: UI/Source/AppKitExtensions/NestableScrollView.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import AppKit
+
+open class NestableScrollView : NSScrollView {
```
Comment (danielrhammond) [inline UI/Source/AppKitExtensions/NestableScrollView.swift]: Totally in favor of pulling this out but seems like we need a change in CollectionViewController to allow for customizing the scrollView used in order for this to be useful within pilot right?

Can just be `makeScrollView` function in the style of `makeRootView`.

Comment (alanjrogers) [reply UI/Source/AppKitExtensions/NestableScrollView.swift]: Yeah good call, the FullWidthScrollView is now a subclass of this (so we're using it by default), but I agree we should add a `makeScrollView` function too.

**Comment by alanjrogers** [comment]
@danielrhammond CR addressed!

File: UI/Source/AppKitExtensions/NestableScrollView.swift (hunk 1 above)
Comment (danielrhammond) [reply UI/Source/AppKitExtensions/NestableScrollView.swift]: Ah. I had missed that it was a subclass. 👍

**Pull request merged**